Explicitly point out not installing ns-cert component for http01 challenge (#2890)
knative/docs: docs/serving/using-auto-tls.md

If you choose to use the mode of provisioning certificate per namespace, you need to install `networking-ns-cert` components.

**Note that provisioning certificate per namespace only works with DNS-01

>  challenge. If you choose to use HTTP-01 challenge, you must NOT install this

Reviewer: Just curious, ns-cert for HTTP-01 is under development? Or it will never support HTTP-01?

Author: It will never support HTTP01 as the namespace-level cert is wildcard cert which can only works with DNS01.

Reviewer: I see. Thank you.

> **Note that provisioning certificate per namespace only works with DNS-01
>  challenge. If you choose to use HTTP-01 challenge, you must NOT install this
>  component**

Reviewer 2: ```suggestion
**IMPORTANT:** Provisioning a certificate per namespace only works with DNS-01
 challenge. This component cannot be used with HTTP-01 challenge.
```

Author: Done

If you choose to use the mode of provisioning certificate per namespace, you need to install `networking-ns-cert` components.

**IMPORTANT:** Provisioning a certificate per namespace only works with DNS-01
 challenge. This component cannot be used with HTTP-01 challenge.

> 

Reviewer 3: @ZhiminXiang can you please also add that:

- HTTP-01 does not support provisioning a certificate per namespace

Up above within the list "1.  Using HTTP-01 challenge" (line 30) to make this clear at the time when choosing DNS-01 vs HTTP-01

Author: Done
